File: lib/src/data/commands_impl/log.dart
import 'package:logging/logging.dart';

import '../../application/commands/log.dart';

final LogReader<Logger> log =
    (type, message) => (logger) => () => switch (type) {
          LogType.debug => logger.fine(message),
          LogType.error => logger.severe(message),
          LogType.info => logger.info(message)
        };

File: src/app/interfaces/write/log.ts
import { io as IO } from 'fp-ts'

export type LogSeverity =
	| 'debug'
	| 'info'
	| 'warning'
	| 'error'

export type Log = (data: {
	severity: LogSeverity
	message: string
}) => IO.IO<void>
